src/components/vm-and-vnic-list.ts

```diff
@@ -21,48 +19,53 @@ export class VmAndVnicListComponent extends paper.Group {
 
   // store lowest vnic point by network name for setting the lowest point of the vapp network path
   private _lowestVnicPointByNetworkName: LowestVnicPointByNetworkName = {};
-  // x position of the last (furthest right) network in vapp network list used for positioning vms and vnics
-  readonly lastNetworkPositionX: number = 0;
 
   /**
    * Creates a new VmAndVnicListComponent instance.
    *
    * @param _vms the vms data
-   * @param vappNetworkPositionsByName the vapp network positions by name from the VappNetworkListComponent for
+   * @param vappNetworkPositionsByName the vapp network positions by name from the VappNetworkListComponent used for
    * positioning x value of matching vnics
+   * @param lastNetworkPosition the position of the last (furthest right) vapp network used for positioning vms and
+   * unattached vnics
    * @param _point the location that the vm and vnic list should be rendered at
    */
   constructor(private _vms: Array<VmData>,
               private vappNetworkPositionsByName: VappNetworkPositionsByName = {},
+              private lastNetworkPosition: paper.Point = new paper.Point(0, 0),
               private _point: paper.Point = new paper.Point(0, 0)) {
     super();
-    this.applyMatrix = false;
     this.pivot = new paper.Point(0, 0);
     this.position = _point;
 
-    // reusable calculations
-    const vappNetworkCount = Object.keys(this.vappNetworkPositionsByName).length;
-    this.lastNetworkPositionX = vappNetworkCount && (vappNetworkCount - 1) * VAPP_NETWORK_RIGHT_MARGIN
-      + CONNECTOR_RADIUS - DEFAULT_STROKE_WIDTH;
-    const vnicOffsetX = vappNetworkCount
-      ? CONNECTOR_SIZE + CONNECTOR_RIGHT_PADDING + DEFAULT_STROKE_WIDTH * 2
-      : VAPP_NETWORK_PADDING_RIGHT;
-    const vnicOffsetY = LABEL_HEIGHT / 2 + VM_MARGIN_VERTICAL;
-    const vmOffsetX = CONNECTOR_RADIUS + CONNECTOR_RIGHT_PADDING + DEFAULT_STROKE_WIDTH * 2;
+    // reusable values
+    const hasVappNetworks = Object.keys(this.vappNetworkPositionsByName).length !== 0;
+    // middle of vm label
+    const vnicOffsetY = LABEL_HEIGHT / 2;
+    // offset from vnic pivot in the center to the left stroke bounds
+    const vnicOffsetX = CONNECTOR_RADIUS - DEFAULT_STROKE_WIDTH;
 
     // draw vms and their vnics
     this._vms.forEach((vmData, index) => {
-      const pointY = (LABEL_HEIGHT + VM_MARGIN_VERTICAL) * index;
+      const sharedPointY = (LABEL_HEIGHT + VM_MARGIN_VERTICAL) * index;
+      let xPositionMultiplier = hasVappNetworks ? 1 : 0;
```

Comment:
this would also work as `let xPositionMultiplier = hasVappNetworks` since it's a boolean, but i wanted to be super explicit about where values come from